axotag/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+#![deny(missing_docs)]
+#![allow(clippy::result_large_err)]
+
+//! # axotag
+//!
+//! This library contains tag-parsing code for use with cargo-dist.
+
+use errors::{TagError, TagResult};
+use semver::Version;
+
+pub mod errors;
+
+/// Represents an opaque package.
+pub struct Package {
+    /// The package's name
+    pub name: String,
+    /// The package's version, if specified
+    pub version: Option<Version>,
+}
+
+/// details on what we're announcing (partially computed)
+pub struct PartialAnnouncementTag {
+    /// The full tag
+    pub tag: String,
+    /// The release
+    pub release: ReleaseType,
+    /// whether we're prereleasing
+    pub prerelease: bool,
+}
+
+impl Default for PartialAnnouncementTag {
+    /// Constructs an empty PartialAnnouncementTag
+    fn default() -> PartialAnnouncementTag {
+        PartialAnnouncementTag {
+            tag: String::new(),
+            release: ReleaseType::None,
+            prerelease: false,
+        }
+    }
+}
+
+/// which type of release we're announcing
+pub enum ReleaseType {
+    /// none
+    None,
```

Comment:
oh interesting... any particular reason this works better than `Option<ReleaseType>`? The latter seems like it would be easier to work with in a lot of places -- especially since I think Announcement wants this to be non-None (i think...? maybe I'm wrong. possibly specifically in the case of needs_coherent = false)